src/poetry/utils/dependency_specification.py

```diff
@@ -22,7 +24,7 @@
     from poetry.utils.env import Env
 
 
-DependencySpec = Dict[str, Union[str, Dict[str, Union[str, bool]], List[str]]]
+DependencySpec = Dict[str, Union[str, bool, Dict[str, Union[str, bool]], List[str]]]
```

Comment:
I think this might deserve promotion to a full type -- at the very least, how about using the new Union syntax (`foo | bar`) for better readability?

Reply:
There is something weird going on with that, even with annotations imported bot ide and type checks complain about that.

Reply:
Odd -- my understanding is that the Python runtime won't evaluate anything inside the brackets until you try to access it at runtime -- it's basically a string. It it mypy that complains, or just PyCharm?

Reply:
Maybe just pycharm; let me double check